redash/query_runner/couchbase.py

```diff
@@ -0,0 +1,172 @@
+import json
+
+logger = logging.getLogger(__name__)
+try:
+    import requests
+    import httplib2
+except ImportError as e:
+    logger.error('Failed to import: ' + str(e))
+
+
+TYPES_MAP = {
+    str: TYPE_STRING,
+    unicode: TYPE_STRING,
+    int: TYPE_INTEGER,
+    long: TYPE_INTEGER,
+    float: TYPE_FLOAT,
+    bool: TYPE_BOOLEAN,
+    datetime.datetime: TYPE_DATETIME,
+    datetime.datetime: TYPE_STRING
+}
+
+
+def _get_column_by_name(columns, column_name):
+    for c in columns:
+        if "name" in c and c["name"] == column_name:
+            return c
+    return None
+
+
+def parse_results(results):
+    rows = []
+    columns = []
+
+    for row in results:
+        parsed_row = {}
+        for key in row:
+            if isinstance(row[key], dict):
+                for inner_key in row[key]:
+                    column_name = u'{}.{}'.format(key, inner_key)
+                    if _get_column_by_name(columns, column_name) is None:
+                        columns.append({
+                            "name": column_name,
+                            "friendly_name": column_name,
+                            "type": TYPES_MAP.get(type(row[key][inner_key]), TYPE_STRING)
+                        })
+
+                    parsed_row[column_name] = row[key][inner_key]
+
+            else:
+                if _get_column_by_name(columns, key) is None:
+                    columns.append({
+                        "name": key,
+                        "friendly_name": key,
+                        "type": TYPES_MAP.get(type(row[key]), TYPE_STRING)
+                    })
+
+                parsed_row[key] = row[key]
+
+        rows.append(parsed_row)
+    return rows, columns
+
+
+class Couchbase(BaseQueryRunner):
+
+    noop_query = 'Select 1'
+
+    @classmethod
+    def configuration_schema(cls):
+        return {
+            'type': 'object',
+            'properties': {
+                'host': {
+                    'type': 'string',
+                },
+                'port': {
+                    'type': 'string',
+                    'title': 'Port (Defaults: 8095 - Analytics, 8093 - N1QL)',
+                    'default': '8095'
+                },
+                'user': {
+                    'type': 'string',
+                },
+                'password': {
+                    'type': 'string',
+                },
+            },
+            'required': ['host', 'user', 'password'],
+            "order": ['host', 'port', 'user', 'password'],
+            "secret": ["password"]
+        }
+
+    def __init__(self, configuration):
+        super(Couchbase, self).__init__(configuration)
+
+    @classmethod
+    def enabled(cls):
+        return True
+
+    @classmethod
+    def annotate_query(cls):
+        return False
+
+    def test_connection(self):
+        result = self.call_service(self.noop_query, '')
+
+    def get_buckets(self, query, nameParam):
```

Comment:
`nameParam` should be `name_param` to be consistent with PEP-8/our naming convention.

Reply:
hi @arikfr. thx for review. Fixed nameParams and merged updates from master - then got test failing and locally message `ImportError: No module named flask_talisman` on `docker-compose` up run.